docs/specs/mlnet-database-loader/mlnet-database-loader-specs.md

```diff
@@ -0,0 +1,407 @@
+We'll need to come up with a reasonably general pattern (probably something that reasons about primary keys), but this scenario is not simple. 
+
+    See [related issue](https://github.com/dotnet/machinelearning-samples/issues/507)
+
+- **Implementation as a separate NuGet package**: Since the implementation of this feature uses the primitives in System.Data, therefore this feature should be available as a separate NuGet package so that ML.NET core packages don't depend on System.Data.
+
+- **Support for sparse data**: The database loader should support sparse data, at least up to the maximum number of columns in SQL Server (1,024 columns per nonwide table, 30,000 columns per wide table or 4,096 columns per SELECT statement).
+
+    ML.NET supports sparse data such as in the following example using a [sparse matrix](https://en.wikipedia.org/wiki/Sparse_matrix) of thousands or even millions of columns even when in this example only 200 columns have real data (sparse data): 
+
+    - [ML.NET sample using millions of columns with sparse data](https://github.com/dotnet/machinelearning-samples/tree/master/samples/csharp/getting-started/LargeDatasets)
+
+    SQL Server supports [Sparse columns](https://docs.microsoft.com/en-us/sql/relational-databases/tables/use-sparse-columns?view=sql-server-2017), however, it is just a way to optimize storage for null values. It still needs to have a real column created in the table per each logical column (i.e. 1,000 columns defined in the SQL table) even when it might not have data.
+
+
+
+
+
+
+# Gated conditions for first Preview release
+
+- TBD
+
+# Gated conditions for GA release
+
+- TBD
+
+# API design examples
+
+The following are a few C# examples on how this API could be used by a developer when loading data from a database into an IDataView.
+
+Whenever in the samples below it is using a `SqlConnection` type, it could also be using any other supported DbConnection in .NET providers, such as:
+
+- `SqlConnection` (SQL Server)
+- `NpgsqlConnection` (PostgreSQL)
+- `OracleConnection` (Oracle)
+- `MySqlConnection` (MySql)
+- etc.
+
+The specific type (`SqlConnection`, `NpgsqlConnection`, `OracleConnection`, etc.) is specified as a generic parameter so references to the needed NuGet packages will be provided by the user when trying to compile.
+
+**1. Data loading from a database table (Simplest approach):**
```

Comment:
Is the differentiation between tables and views etc. necessary in the API? Naively, I would have expected to provide a SQL Query which returns tuples of a schema matching the `IDataView` to create. Whether that query is `SELECT * FROM [TABLE]` or anything else would not matter then, right?

Reply:
You are right, differentiation between tables and views etc. are not necessary in the API, they are just convenient simpler methods since it'll be very common to consolidate a prepared dataset in a single table. But I agree that having the SQL Query sentence method can also be enough.

The 'LoadFromDbTable()' or 'LoadFromDbView()' methods are just "sugar methods" re-using the other two foundational methods.

So, yeah, I find those methods optional and we could provide them or not depending on customer feedback.